resolvewithplus.js

```diff
@@ -318,7 +318,9 @@ const esmparse = (spec, specifier, opts = {}) => {
           // if dynamic 'spectype', lookup 'commonjs' or 'module'
           // according to package.json
           specname = specname === spectype
-            ? getspectypenamedexportdefault(opts.packagejsontype)
+            ? isDirPathRe.test(specifier)
```

Comment:
the whole logic looks a bit... suspicious for me - it's a kind of mix of 'how to resolve exact import' (exports field logic) and 'how to detect module type - CJS or ESM' (type field). but, if it works - let's merge

Reply:
agree :/